src/styles/index.scss

```diff
@@ -1,3 +1,74 @@
 body {
   margin: 0;
 }
+
+$days: (
+  mon: 1,
+  tue: 2,
+  wed: 3,
+  thu: 4,
+  fri: 5,
+  sat: 6,
+  sun: 7,
+);
+$durations: 28, 29, 30, 31;
+
+.calendar {
+  width: 707px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  padding: 10px;
+  margin-top: 72px;
+  margin-inline: 149px;
+
+  @each $name, $pos in $days {
+    &--start-day-#{$name} {
+      .calendar__day:first-child {
+        margin-left: ($pos - 1) * 101;
+      }
+    }
+  }
+
+  //           On hovering over a cell:
+  // - cursor should become pointer
+  // - The hovered cell has to become pink (use `#FFBFCB`)
+  // - Move the hovered cell up by `20px` (use `transform`)
+  // - All changes should be animated with the duration of 0.5s
+
+  @each $month-duration in $durations {
+    // modifiers for possible month durations
+    &--month-length-#{$month-duration} {
+      @for $day from 1 through $month-duration {
+        // draw correct number of dates cells
+        & .calendar__day:nth-child(#{$day}) {
+          position: relative;
+          box-sizing: border-box;
+          background-color: #eee;
+          height: 100px;
+          width: 100px;
+          border: 1px solid #000;
+          transition-duration: 500ms;
+
+          // draw dates numbers
+          &::before {
+            content: '#{$day}';
+            position: absolute;
+            font-family: Arial, Helvetica, sans-serif;
+            font-size: 30px;
+            top: 50%;
+            left: 50%;
+            transform: translate(-50%, -50%);
+            text-align: center;
+          }
+
+          &:hover {
+            cursor: pointer;
+            background-color: #ffbfcb;
+            transform: translateY(-20px);
+          }
```

Comment:
The hover effects are implemented correctly. Ensure that these styles are applied by linking the correct SCSS file in the HTML.